libraries/AP_Math/examples/matrix_alg/matrix_alg.cpp

```diff
@@ -0,0 +1,73 @@
+/// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+//
+// Unit tests for the AP_Math rotations code
+//
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+static void test_matrix_inverse(void)
+{
+    //fast inverses
+    float mat5x5[] = {1.0f, 2.0f, 0.3f, 0.4f, 0.3f,   \
+                      0.5f, 6.0f, 0.7f, 0.8f, 0.4f,   \
+                      0.9f, 1.0f, 1.1f, 1.2f, 0.5f,   \
+                      1.3f, 1.4f, 1.5f, 1.6f, 0.6f,   \
+                      0.5f, 0.7f, 0.7f, 0.8f, 0.4f};
+
+    float mat[25];
+    if(inverse(mat5x5,mat,3)){
```

Comment:
That's not a test suite! The test would pass fine if inverse() implemented a transpose or a negation. Please at least do the obvious test of multiplying a matrix by its inverse, and please generate a pile of random matrices